File: Week-2/Task 4.py
def get_number(index):
    # your code here

    # set initial number
    tuple_list = [(0,4)]

    # calculate how long the list is  
    lenth_tuple_list = None
    if (index + 1) % 2 == 0:
        lenth_tuple_list = int((index + 1) / 2)
    else:
        lenth_tuple_list = (index + 1)//2 + 1

    # create list 
    for n in range(1,lenth_tuple_list):
        tuple_last_term = tuple_list[n-1]
        tuple_add = (tuple_last_term[0]+3, tuple_last_term[1]+3)
        tuple_list.append(tuple_add)

    # find number
    index_in_tuple = None
    if index % 2 == 0:
        index_in_tuple = 0
    else:
        index_in_tuple = 1
    
    tuple_chosen = tuple_list[lenth_tuple_list-1]
    num_chosen = tuple_chosen[index_in_tuple]

    print(num_chosen)


get_number(1) # print 4
get_number(5) # print 10
get_number(10) # print 15

File: Week-2/Task 4.js
function getNumber(index){
    // your code here
    
    // set initial number
    const totalArray = [[0,4]];

    // calculate how long the list is  
    let lenArray = 0;
    if ( Number.isInteger((index + 1) / 2)) {
        lenArray = (index + 1) / 2;
    }else{
        lenArray = Math.round((index + 1) / 2);
    }

    // create list 
    for ( let i = 1; i < lenArray; i++ ) {
        let last_term = totalArray[i-1];
        let element_add = [ last_term[0]+3 , last_term[1]+3];
        totalArray.push(element_add);
    };
    
    // find number
    let index_in_element = null;
    if ( Number.isInteger(index / 2)) {
        index_in_element = 0;
    }else{
        index_in_element = 1;
    }

    let array_chosen = totalArray[lenArray-1];
    let num_chosen = array_chosen[index_in_element];

    console.log(num_chosen);
}
    getNumber(1); // print 4
    getNumber(5); // print 10
    getNumber(10); // print 15